packages/blade/src/components/Preview/Preview.web.tsx

```diff
@@ -0,0 +1,369 @@
+      </BaseBox>
+      {trailing}
+    </BaseBox>
+  );
+};
+
+const PreviewFooter = assignWithoutSideEffects(_PreviewFooter, {
+  componentId: MetaConstants.PreviewFooter,
+});
+
+const dotSpacing = 16;
+const dotOpacity = 0.1;
+const dotSize = 1;
+const ZoomContainer = styled.div<{ isDragEnabled: boolean; isDragging: boolean }>`
+  width: 100%;
+  height: 100%;
+  cursor: ${({ isDragEnabled, isDragging }) => {
+    if (!isDragEnabled) return 'default';
+    return isDragging ? 'grabbing' : 'grab';
+  }};
+  background-image: radial-gradient(
+    circle,
+    rgba(0, 0, 0, ${dotOpacity}) ${dotSize}px,
+    transparent ${dotSize}px
+  );
+  background-size: ${dotSpacing}px ${dotSpacing}px;
+  .zoom-wrapper,
+  .zoom-content {
+    width: 100%;
+    height: 100%;
+    overflow: visible;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+  }
+  transition: cursor 0.1s ease;
+`;
+
+const TransFormWrapperContainer = styled.div`
+  width: 100%;
+  height: 100%;
+  position: relative;
+  overflow: hidden;
+  background-color: ${({ theme }) => theme.colors.surface.background.gray.moderate};
+`;
+const Preview = ({
+  children,
+  onFullScreen: onFullScreenProp,
+  onZoomChange,
+  zoomScaleStep = 0.1,
+  isDragAndZoomDisabled = false,
+  defaultZoom,
+  onDragChange,
+}: PreviewProps): React.ReactElement => {
+  const [controlledZoom, setControlledZoom] = useControllableState({
+    onChange: onZoomChange,
+    defaultValue: defaultZoom ?? 1,
+  });
+  const [isFullScreen, setIsFullScreen] = useState(false);
+  const [isDragging, setIsDragging] = useState(false);
+  const containerRef = React.useRef<HTMLDivElement>(null);
+
+  const handleTransformed = ({
+    state,
+  }: {
+    state: { scale: number; positionX: number; positionY: number };
+  }): void => {
+    const { scale, positionX, positionY } = state;
+    setControlledZoom(() => scale);
+    onDragChange?.({ x: positionX, y: positionY });
+  };
+
+  const handleFullScreenError = (err: unknown): void => {
+    if (err instanceof DOMException) {
+      console.error(`Fullscreen request failed: ${err.name} - ${err.message}`);
+    } else {
+      console.error('Unexpected error during fullscreen request:', err);
+    }
+  };
+
+  const handleFullScreen = useCallback(async () => {
+    if (!document.fullscreenElement) {
+      try {
+        await containerRef.current?.requestFullscreen();
+        setIsFullScreen(true);
+        onFullScreenProp?.();
+      } catch (err: unknown) {
+        handleFullScreenError(err);
+      }
+    } else {
+      try {
+        await document.exitFullscreen();
+        setIsFullScreen(false);
+        onFullScreenProp?.();
+      } catch (err: unknown) {
+        handleFullScreenError(err);
+      }
+    }
+  }, [onFullScreenProp]);
+
+  // Handle keyboard shortcut for fullscreen
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent): void => {
+      if ((event.metaKey || event.ctrlKey) && event.key === 'f') {
```

Comment:
Why are we hijacking cmd+f (search in page) shortcut?

Reply:
i think we should just keep it "F" , since youtube also have that as a shortcut

Reply:
Right now the shortcut seems to be on the `document`. Doesn't that mean it will end up full-screening the preview while I am typing something in the input 🙈?

I don't think so we should have shortcut key for full screening. In case of youtube or vimeo it makes sense because the media is the focus for those sites 🤔